ethcore/Cargo.toml

```diff
@@ -17,9 +17,9 @@ ethash = { path = "../ethash" }
 ethcore-bloom-journal = { path = "../util/bloom" }
 ethcore-bytes = { path = "../util/bytes" }
 fetch = { path = "../util/fetch" }
-hashdb = { path = "../util/hashdb" }
-memorydb = { path = "../util/memorydb" }
-patricia-trie = { path = "../util/patricia_trie" }
+hashdb = { version = "0.2", path = "../util/hashdb" }
+memorydb = { version = "0.2", path = "../util/memorydb" }
+patricia-trie = { version = "0.2", path = "../util/patricia_trie" }
```

Comment:
`version =` is not needed, unless you want to publish `ethcore` to crates.io. In all other cases, it's always ignored if it occurs together with `path =`

Reply:
Ok, that makes sense (not sure how it'd work otherwise). The idea here was mainly to signal that there are breaking changes.

Maybe this is a good time to pause and figure out what the versioning strategy should be here. When this work started we were planning to extract `patricia-trie` into its own crate outside the parity repo. This was abandoned (for good reason), but we never devised a precise plan for sharing the code with polkadot.  I think we have two options for this.

1. Leave everything in-tree and add a new `ethereum-patricia-trie` crate as @rphmeier suggests that acts as a facade to `patricia-trie` (reexports, convenience type aliases and whatnot). Add mini-crates for `KeccakHasher` and `RlpNodeCodec`.

2. Publish crates to crates.io from inside the parity repo and let parity use `v0.1` for as long as it needs to but ideally not indefinitely – avoiding a fork is the whole point of this work; polkadot can use `v0.2`. We'd still need a facade of some kind but it could potentially live inside the `patricia-trie` crate itself temporarily.

I think option 2 sounds like a maintenance nightmare and I'd like to go with option 1 (or hear about alternatives!).